cms/nextjs-contentful-store/pages/[countryCode]/[lang]/[product].tsx

```diff
@@ -0,0 +1,199 @@
+            <p className="text-gray-900 text-3xl title-font font-medium my-3">{product.name}</p>
+            <p className="text-gray-600 text-xl title-font font-medium my-3">{selectedVariant}</p>
+            <p className="leading-relaxed">{product.description}</p>
+            <div className="flex items-center border-b-2 border-gray-200 py-5">
+              <div className="flex items-center">
+                <div className="relative" data-children-count="1">
+                  <select
+                    placeholder={locale[lang].selectSize as string}
+                    className="rounded border appearance-none border-gray-400 py-2 focus:outline-none focus:border-blue-500 text-base pl-3 pr-10"
+                    value={selectedVariant}
+                    onChange={(e) => setSelectedVariant(e.target.value)}
+                  >
+                    {variantOptions?.map((option) => (
+                      <option key={option.code} value={option.code}>
+                        {option.label}
+                      </option>
+                    ))}
+                  </select>
+                  <span className="absolute right-0 top-0 h-full w-10 text-center text-gray-600 pointer-events-none flex items-center justify-center">
+                    <svg
+                      fill="none"
+                      stroke="currentColor"
+                      strokeLinecap="round"
+                      strokeLinejoin="round"
+                      strokeWidth="2"
+                      className="w-4 h-4"
+                      viewBox="0 0 24 24"
+                    >
+                      <path d="M6 9l6 6 6-6"></path>
+                    </svg>
+                  </span>
+                </div>
+              </div>
+            </div>
+            <div className="flex justify-between items-center pt-5">
+              <span className="title-font font-medium text-2xl text-gray-900">
+                <PricesContainer>
+                  <Price
+                    skuCode={selectedVariant}
+                    className="text-indigo-600 mr-1"
+                    compareClassName="text-gray-500 line-through text-lg"
+                  />
+                </PricesContainer>
+              </span>
+              <AddToCartButton
+                skuCode={selectedVariant}
+                label={locale[lang].addToCart as string}
+                className="inline-flex items-center px-4 py-2 border border-gray-300 shadow-sm text-sm md:text-base font-medium rounded-md text-white bg-gray-900 hover:bg-gray-800 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500 disabled:opacity-50 disabled:cursor-not-allowed"
+              />
+            </div>
+          </div>
+        </div>
+      </div>
+    </Page>
+  );
+};
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  return {
+    paths: [],
+    fallback: true
+  };
+};
+
+export const getStaticProps: GetStaticProps = async ({ params }: any) => {
```

Comment:
Here you can set the `Props` the component expects to receive.

```suggestion
export const getStaticProps: GetStaticProps<Props> = async ({ params }: any) => {
```